src/actions/compass.ts

```diff
@@ -0,0 +1,45 @@
+import {SpecQueryModelGroup} from 'compassql/build/src/model';
+import {Query} from 'compassql/build/src/query/query';
+import {Schema} from 'compassql/build/src/schema';
+import {Dispatch} from 'redux';
+import {ThunkAction} from 'redux-thunk';
+import {ReduxAction} from './redux-action';
+
+import {fetchCompassRecommends} from '../api/api';
+import {State} from '../models/index';
+import {getQuery, getSchema} from '../selectors';
+import {Action} from './index';
+
+export type CompassAction = CompassRecommendsRequest | CompassRecommendsRecieve;
+export type CompassAsyncAction = CompassRecomendationsLoad;
+
+export const COMPASS_RECOMMENDS_REQUEST = 'COMPASS_RECOMMENDS_REQUEST';
+export type CompassRecommendsRequest = ReduxAction<typeof COMPASS_RECOMMENDS_REQUEST, {}>;
+
+export const COMPASS_RECOMMENDS_RECEIVE = 'COMPASS_RECOMMENDS_RECEIVE';
+export type CompassRecommendsRecieve = ReduxAction<typeof COMPASS_RECOMMENDS_RECEIVE, {
+  recommends: SpecQueryModelGroup
+}>;
+
+export type CompassRecomendationsLoad = ThunkAction<void , State, undefined>;
+export function compassRecomendationsLoad(query?: Query, schema?: Schema): CompassRecomendationsLoad {
```

Comment:
I enjoyed this article from mapbox: https://www.mapbox.com/blog/redux-for-state-management-in-large-web-apps/

which had this to say about action creator names:

> Our action creators are functions named with simple imperative verb phrases, e.g. updateCreditCard, verifyEmail, uploadDataset, dismissWelcomeModal . Often these function names do not correspond to technical plumbing behind the scenes; instead, they describe an intention, a command. The action creator might actually dispatch several actions, depending on what it must accomplish. The point is that it does what you ask it to do.